src/metrics.go

```diff
@@ -10,122 +15,204 @@ import (
 	columnFamilyMetrics := make(map[string]map[string]interface{})
-	visitedColumnFamilies := make(map[string]struct{})
 
-	re, err := regexp.Compile("keyspace=(.*),scope=(.*?),")
+	columnFamilyQueryConfig, err := getColumnFamilyQueries(client, queryConfig)
 	if err != nil {
-		return nil, nil, err
+		return nil, fmt.Errorf("failed to fetch 'column-family' metrics: %w", err)
 	}
 
-	for _, query := range jmxPatterns {
-		results, err := client.QueryMBeanAttributes(query)
+	for _, query := range columnFamilyQueryConfig {
+		attrNames := query.GetAttributeNames()
+
+		results, err := client.GetMBeanAttributes(query.MBean, attrNames...)
 		if err != nil {
 			if jmxErr, ok := gojmx.IsJMXError(err); ok {
-				log.Debug("Error querying %s: %v", query, jmxErr)
+				log.Debug("Failed to get 'column-family' attributes: %s, for mBeanName %s: %v", attrNames, query.MBean, jmxErr)
 				continue
 			}
-			return nil, nil, fmt.Errorf("fatal jmx error while querying: %q: %w", query, err)
+			return nil, fmt.Errorf("failed to fetch 'column-family' metrics, query: %q: attributes: %s error: %w", query.MBean, attrNames, err)
 		}
 
 		for _, jmxAttr := range results {
 			if jmxAttr.ResponseType == gojmx.ResponseTypeErr {
-				log.Debug("Failed to process attribute for query: %s status: %s", jmxAttr.Name, jmxAttr.StatusMsg)
+				log.Debug("Failed to retrieve 'column-family' attribute for query: %s status: %s", jmxAttr.Name, jmxAttr.StatusMsg)
 				continue
 			}
-			matches := re.FindStringSubmatch(jmxAttr.Name)
-			key := re.ReplaceAllString(jmxAttr.Name, "")
-
-			if len(matches) != 3 {
-				metrics[key] = jmxAttr.GetValue()
-			} else {
-				columnfamily := matches[2]
-				keyspace := matches[1]
-				eventkey := keyspace + "." + columnfamily
-
-				_, found := internalKeyspaces[keyspace]
-				if !found {
-					_, found := visitedColumnFamilies[eventkey]
-					if !found {
-						if len(visitedColumnFamilies) < args.ColumnFamiliesLimit {
-							visitedColumnFamilies[eventkey] = struct{}{}
-						} else {
-							log.Warn("Skipping column family %s due to limit reached. Current limit set to %d",
-								columnfamily, args.ColumnFamiliesLimit)
-							continue
-						}
-					}
-
-					_, ok := columnFamilyMetrics[eventkey]
-					if !ok {
-						columnFamilyMetrics[eventkey] = make(map[string]interface{})
-						columnFamilyMetrics[eventkey]["keyspace"] = keyspace
-						columnFamilyMetrics[eventkey]["columnFamily"] = columnfamily
-						columnFamilyMetrics[eventkey]["keyspaceAndColumnFamily"] = eventkey
-					}
-					columnFamilyMetrics[eventkey][key] = jmxAttr.GetValue()
-				}
 
+			matches := columnFamilyRegex.FindStringSubmatch(jmxAttr.Name)
+			key := columnFamilyRegex.ReplaceAllString(jmxAttr.Name, "")
+
+			columnFamily := matches[2]
+			keyspace := matches[1]
+			eventKey := keyspace + "." + columnFamily
+
+			_, ok := columnFamilyMetrics[eventKey]
```

Comment:
would make sense to use `NEWVAR,ok:=columnFamilyMetrics[eventKey]` and then `NEWVAR` as a local variable? just to improve the reading experience

Reply:
oh maybe not, because the biggest section using it is in case that does not exist